src/Hedgehog/Gen.fs

```diff
@@ -490,15 +490,20 @@ module Gen =
         toRandom g
         |> Random.run seed 30
 
+    let asString (g : Gen<'a>) : string =
```

Comment:
This looks good but I believe it can also be given some more thought. 

I would expect most users to understand that *print* + *sample* does exactly that:
* *print* (to the console) a *sample* of `g`

Now, if we introduce `asString` then it *may* cause confusion (IME a small % of the actual users are following each and every PR). So as a user that just upgraded to latest Hedgehog I would expect that `asString`:
* it returns a string as `g` in its fully qualified name?
* it returns a string representing some structure of `g` prior being run/sampled/shrinked?

---

Perhaps *sample* is the name we would want, after all?

Reply:
`sample` is already defined, though. :disappointed:

Reply:
I suggest we remove this function (called `Gen.printSample` in `master` and `Gen.asString` in this PR).

Instead, move the body of the `for` loop to a new function called `Tree.asString`.  Then users can easily implement this function themselves if they really want to see exactly five shrink tree printed to standard out.

Reply:
I renamed these functions to `format`, `formatWith`, and `formatSample`.